flink-runtime/src/main/java/org/apache/flink/runtime/state/heap/NestedMapsStateTable.java

```diff
@@ -350,11 +353,13 @@ public int sizeOfNamespace(Object namespace) {
 		private final StateSnapshotTransformer<S> snapshotFilter;
 
 		NestedMapsStateTableSnapshot(
-			NestedMapsStateTable<K, N, S> owningTable, StateSnapshotTransformFactory<S> snapshotTransformFactory) {
+			NestedMapsStateTable<K, N, S> owningTable,
+			StateSnapshotTransformFactory<S> snapshotTransformFactory,
+			TypeSerializer<K> keySerializer) {
 
 			super(owningTable);
 			this.snapshotFilter = snapshotTransformFactory.createForDeserializedState().orElse(null);
-			this.keySerializer = owningStateTable.keyContext.getCurrentKeySerializer();
+			this.keySerializer = keySerializer;
```

Comment:
can't we just do `this.keySerializer = owningStateTable.keySerializer`, and avoid passing in the key serializer when instantiating `NestedMapsStateTableSnapshot`?